src/shared_modules/utils/tests/stringHelper_test.cpp

```diff
@@ -260,3 +261,55 @@ TEST_F(StringUtilsTest, findRegexInStringExtractingSecondGroup)
     EXPECT_TRUE(Utils::findRegexInString(valueToCheck, matchedValue, regex, 2));
     EXPECT_EQ(matchedValue, "extracted");
 }
+
+TEST_F(StringUtilsTest, convertToUTF8NoChanges)
+{
+    std::string noUnicodeString{"This is a test"};
+    Utils::ISO8859ToUTF8(noUnicodeString);
+    EXPECT_EQ("This is a test", noUnicodeString);
+}
+
+TEST_F(StringUtilsTest, rawUnicodeToUTF8)
+{
+    std::stringstream fileContent;
+    fileContent << R"(CLASSES=none
+BASEDIR=/opt/csw
+INSTDATE=Jan 09 2023 14:35
+PKGSAV=/var/sadm/pkg/CSWschilybase/save
+PKGINST=CSWschilybase
+PSTAMP=joerg@unstable9x-20130619141117
+EMAIL=joerg@opencsw.org
+HOTLINE=http://www.opencsw.org/bugtrack/
+VENDOR=http://cdrecord.berlios.de/old/private/  packaged for CSW by Jörg Schilling
+CATEGORY=application
+NAME=schilybase - A collection of common files from J. Schilling
+PKG=CSWschilybase
+VERSION=1.01,REV=2013.06.19
+ARCH=i386
+OAMBASE=/usr/sadm/sysadm
+PATH=/sbin:/usr/sbin:/usr/bin:/usr/sadm/install/bin
+TZ=localtime
+LANG=C
+LC_ALL=
+LC_MONETARY=
+LC_MESSAGES=
+LC_COLLATE=
+LC_TIME=
+LC_NUMERIC=
+LC_CTYPE=)";
+
+    std::string content;
+
+    while (fileContent.good())
+    {
+        std::string line;
+        std::getline(fileContent, line);
+        // Convert 'line' to UTF-8
+        Utils::ISO8859ToUTF8(line);
+
+        content += line;
+    }
+
+    EXPECT_EQ("CLASSES=noneBASEDIR=/opt/cswINSTDATE=Jan 09 2023 14:35PKGSAV=/var/sadm/pkg/CSWschilybase/savePKGINST=CSWschilybasePSTAMP=joerg@unstable9x-20130619141117EMAIL=joerg@opencsw.orgHOTLINE=http://www.opencsw.org/bugtrack/VENDOR=http://cdrecord.berlios.de/old/private/  packaged for CSW by J\xC3\x83\xC2\xB6rg SchillingCATEGORY=applicationNAME=schilybase - A collection of common files from J. SchillingPKG=CSWschilybaseVERSION=1.01,REV=2013.06.19ARCH=i386OAMBASE=/usr/sadm/sysadmPATH=/sbin:/usr/sbin:/usr/bin:/usr/sadm/install/binTZ=localtimeLANG=CLC_ALL=LC_MONETARY=LC_MESSAGES=LC_COLLATE=LC_TIME=LC_NUMERIC=LC_CTYPE=",
```

Comment:
Could we split this long line?

Reply:
On the other side, the ISO-8859-1 character `ö` has a value of `\xF6`.

Shouldn't the UTF-8 equivalent have only the 2 bytes `\xC3\xB6` instead of the four bytes `\xC3\x83\xC2\xB6` ?

See
``` bash
#!/usr/bin/env python3

c = chr(0xf6)
print(c)
print(c.encode('utf-8'))
print(c.encode('iso-8859-1'))

```

Reply:
Good catch, the stringbuffer is UTF-8